src/algebra/hom/iterate.lean

```diff
@@ -164,8 +164,30 @@ smul_iterate (mul_opposite.op a) n
 lemma mul_right_iterate_apply_one : (* a)^[n] 1 = a ^ n :=
 by simp [mul_right_iterate]
 
+@[simp, to_additive]
+lemma pow_iterate (n : ℕ) (j : ℕ) : ((λ (x : G), x^n)^[j]) = λ x, x^(n^j) :=
+begin
+  induction j with j ih, { simpa, },
+  ext,
+  simp only [iterate_succ, comp_app, ih, ← pow_mul, pow_succ],
+end
+
 end monoid
 
+section group
+
+variables [group G]
+
+@[simp, to_additive]
+lemma zpow_iterate (n : ℤ) (j : ℕ) : ((λ (x : G), x^n)^[j]) = λ x, x^(n^j) :=
+begin
+  induction j with j ih, { simpa, },
```

Comment:
I could avoid using induction by starting out with `obtain ⟨k, rfl | rfl⟩ := n.eq_coe_or_neg,` and factoring through `pow_iterate` but I think it's better not to do this.